tests/phpunit/elementor/core/upgrade/test-upgrades.php

```diff
@@ -430,6 +431,26 @@ public function test_v_3_8_0_fix_php8_image_custom_size() {
 		$this->delete_image( $attachment_id );
 	}
 
+	public function test_v_3_16_0_container_updates() {
+
+		Plugin::$instance->experiments->set_feature_default_state( 'container', 'active' );
+		//Plugin::$instance->experiments->set_feature_default_state( 'nested-elements', 'active' );
+
+		$documents = [];
+		$updater = $this->create_updater();
+
+		$documents[] = $this->create_document_with_data( Test_Module::$document_mock_default_with_container );
+		$documents[] = $this->create_document_with_data( Test_Module::$document_mock_default );
+		$documents[] = $this->create_document_with_data( Test_Module::$document_mock_nested_tabs );
+
+		Upgrades::_v_3_16_0_container_updates( $updater );
+
+		$this->assert_container_changed( $this->get_decoded_post_meta( $documents[0] ) );
+		$this->assert_sections_not_changed( $this->get_decoded_post_meta( $documents[1] ) );
```

Comment:
@TzviRabinovitch Can you maybe use `$document->get_json_meta( '_elementor_data' );` here instead?

Reply:
doing that throws the same error as in adding the test with the nested tabs 
we will try to fix it in the separate task